app/controllers/items_controller.rb

```diff
@@ -209,6 +209,15 @@ def datastream_update
       fail ArgumentError, 'XML is not well formed!'
     end
     @object.datastreams[params[:dsid]].content = params[:content] # set the XML to be verbatim as posted
+
+    # Catch reindexing errors here to avoid cascading errors
+    begin
+      save_and_reindex
+    rescue ActiveFedora::ObjectNotFoundError => e
+      render text: 'The object was not found in Fedora. Please recheck the RELS-EXT XML.', status: :not_found
```

Comment:
are there other situations (besides the one addressed by this PR) where `ActiveFedora::ObjectNotFoundError` could get thrown here?

also, could the wording be a bit more specific?  maybe "An object was not found in Fedora. Possible bad object reference in RELS-EXT XML?"

or, if a bad reference really is the only realistic way this path gets hit, maybe "An object was not found in Fedora. Please check for invalid references in RELS-EXT XML."